baseclasses/chemical_energy/potentiometry_gaschromatography.py

```diff
@@ -0,0 +1,478 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+
+from nomad.metainfo import (Quantity, Reference, Section, SubSection, Datetime, MEnum)
+from nomad.datamodel.data import ArchiveSection
+from nomad.datamodel.metainfo.plot import PlotSection
+
+from .. import BaseMeasurement
+from baseclasses.chemical_energy import CENECCElectrodeID
+
+class NECCFeedGas(ArchiveSection):
+
+    name = Quantity(
+        type=str,
+        shape=[],
+        a_eln=dict(
+            component='EnumEditQuantity',
+            props=dict(
+                suggestions=[
+                    'CO2',
+                    'CO',
+                    'H2'
+                ])))
+
+    flow_rate = Quantity(
+        type=np.dtype(np.float64),
+        unit=('ml/minute'),
+        a_eln=dict(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='ml/minute',
+            props=dict(minValue=0)
+        ))
+
+class NECCExperimentalProperties(ArchiveSection):
+
+    experiment_id = Quantity(
+        type=str,
+        description='This ID is generated automatically in the format User_Date_Number.',
+        a_eln=dict(component='StringEditQuantity', label='Experiment id'))
+
+    # TODO reference the ID here or the CENECCElectrode?
+    anode_id = Quantity(
+        type=Reference(CENECCElectrodeID.m_def),
+        a_eln=dict(component='ReferenceEditQuantity'))
+
+    cathode_id = Quantity(
+        type=Reference(CENECCElectrodeID.m_def),
+        a_eln=dict(component='ReferenceEditQuantity'))
+
```

Comment:
* Do I use the ElectrodeID or the CENECCElectrode class for referencing in the properties?

Reply:
for this there is an extra class from nomad which is i think is best:
```python
 electrode = SubSection(
        section_def=CompositeSystemReference,
    )
```
This solves the issue of putting the id or the reference
if you put in the id it actually searches for the archive and references it!

Reply:
Thank you. :white_check_mark: